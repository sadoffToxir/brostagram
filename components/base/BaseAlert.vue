<template>
  <v-container>
    <v-alert
      :data-test="componentTag"
      :type="type"
      :variant="variant"
      :title="title"
      :text="text"
      :icon="icon"
      :closable="closable"
      :close-label="closeLabel"
    />
  </v-container>
</template>

<script setup lang="ts">

interface Props {
  type?: 'error' | 'success' | 'warning' | 'info'
  variant?: 'flat' | 'text' | 'elevated' | 'tonal' | 'outlined' | 'plain'
  icon?: string | false | undefined
  title: string
  text: string
  closable?: boolean
  closeLabel?: string
}

withDefaults(defineProps<Props>(), {
  type: 'info',
  variant: 'flat',
  icon: undefined,
  closable: false,
  closeLabel: 'Close'
})

const componentTag = 'BaseAlert'

</script>
