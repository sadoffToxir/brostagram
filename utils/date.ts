type CustomDate = string | number | Date

export const formatToIsoDate = (date: CustomDate) => new Date(date).toISOString().split('T')[0]
export const formatToIsoDateAndTime = (date: CustomDate) => new Date(date).toISOString().split('T')[0] + ' at ' + new Date(date).toISOString().split('T')[1].substring(0, 5)
