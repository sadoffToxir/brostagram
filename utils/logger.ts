import { WritableComputedRef, Ref } from 'vue'

export default function logger (consoleText: string, observed: WritableComputedRef<any> | Ref<any>) {
  const config = useRuntimeConfig()

  if (config.public.environment !== 'production') {
    watch(observed, useDebounce(() => {
    // eslint-disable-next-line no-console
      console.log(`${consoleText}:`, useCloneDeep(observed.value))
    }, 500), { deep: true })
  }
}
