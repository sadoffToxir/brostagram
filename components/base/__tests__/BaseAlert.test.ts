import { describe, it, expect } from 'vitest'
import BaseAlert from '../BaseAlert.vue'
import { renderWithVuetify, screen } from '@/tests/vueTestingLibrary/utils'

const componentTag = 'BaseAlert'

describe('BaseAlert', () => {
  const setup = (props?: any) => renderWithVuetify(BaseAlert, {
    props: {
      title: 'Test title',
      text: 'Test text',
      ...props
    }
  })

  it('should render with default props', () => {
    setup()

    const element = screen.getByTestId(componentTag)

    expect(element).toBeInTheDocument()
    expect(element).toMatchSnapshot()
  })

  it('should render with custom props', () => {
    const title = 'Custom title'
    const text = 'Custom text'
    const classes = [
      'text-success',
      'v-alert--variant-outlined'
    ]
    const props = {
      title,
      text,
      variant: 'outlined',
      type: 'success'
    }
    setup(props)

    const element = screen.queryByTestId(componentTag)

    expect(element).toHaveTextContent(title)
    expect(element).toHaveTextContent(text)
    expect(element).toHaveClass(...classes)
  })
})
