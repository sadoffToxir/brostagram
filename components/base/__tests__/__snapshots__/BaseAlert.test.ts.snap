// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseAlert > should render with default props 1`] = `
<div
  class="v-alert v-theme--light bg-info v-alert--density-default v-alert--variant-flat"
  data-test="BaseAlert"
  role="alert"
>
  
  <!---->
  <span
    class="v-alert__underlay"
  />
  
  <!---->
  <div
    class="v-alert__prepend"
  >
    <i
      aria-hidden="true"
      class="mdi-information mdi v-icon notranslate v-theme--light"
      density="default"
      style="font-size: 28px; height: 28px; width: 28px;"
    />
  </div>
  <div
    class="v-alert__content"
  >
    <div
      class="v-alert-title"
    >
      Test title
    </div>
    Test text
    <!---->
  </div>
  <!---->
  <!---->
</div>
`;
